docs/manual/developer/04_style_guide.md

```diff
@@ -0,0 +1,229 @@
+* `rear-matter` (HTML-Like)
+* `version`
+
+### Controls
+These rules apply to the files in `*.yml`
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+
+### Control Sections
+Control sections must be in the following order, if present.
+* `policy`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `id`
+  * Must be short
+  * Must be lowercase
+  * If product specific should be in the format `standard_product`. For example CIS on RHEL8 would be `cis_rhel8`
+  * Should match the filename of the control
+  * Words shall be separated by an underscore
+* `version`
+  * Should be the same as the standard
+* `source`
+  * URL to the standard
+* `controls`
+  * `id`
+  * `levels`
+    * Should be in lowercase
+    * Must have words separated by an underscore
+    * Shall follow the standard
+  * `title`
+    * Shall be one line
+  * `status`
+  * `notes`
+    * Must be a block
+  * `rules`
+    * Must be a valid rule id
+
+## Remediation
+All remediations should have the following header with the appropriate values 
+```bash
+# platform = multi_platform_all
+# reboot = false
+# strategy = enable
+# complexity = low
+# disruption = low
+```
+
+### Bash
+* Should use macros instead shared functions
+* Shall use 4-space indentation
+* Put `do` or `then` on the same line as `for` or `if` respectively, e.g. `for file in *; do`
+
+### Ansible
+* Shall follow all the rules in the [YAML](manual/developer/04_style_guide:yaml) section
+* Should prefer using Ansible modules over just calling system commands
+* Shall be written to pass `ansible-lint`
+
+### Kubernetes
+* Shall follow all the rules in the [YAML](manual/developer/04_style_guide:yaml) section
+
+## XML
+* Shall use the `.xml` for the file extension
+* Must be able to be parsed by Python's XML parser
+* Shall use 4-space indentation
+
+### OVAL
+* The `id` attribute of `<definition>` should be `{{{ rule_id }}}`
+* The elements should be in the following order:
+  * `def-group`
+    * `definition`
+      * `metadata`
+        * Should be defined by the`oval_metadata` macro
+      * `criteria`
+    * Like OVAL `test`, `object`, and `state` should have the same name after their respective prefix.
+    * Like OVAL `test`, `object`, and `state` should group together, if there are many tests the order below should be repeated for each group.
+      * `test`
+      * `object`
+      * `state`
+
+### Test Elements
+* `id` should start with `test_`
+
+### Object Elements
+* `id` should start with `obj_`
+
+### State Elements
+* `id` should start with `state_`
+
+## Tests
+* Shall always do something, even if testing default behavior
+* Shall test one change
+* Shall use the `.sh` for the file extension
+* Shall use the `#!/bin/bash` shebang at the first line
+
+## Markdown
+* [One sentence per line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line)
+* Shall use the `.md` for the file extension
```

Comment:
The link is for asciidoc which is a different format than markdown

Reply:
This is true, but the ideas are similar @matejak suggested this link.